lms/static/scripts/frontend_apps/utils/onedrive-picker-client.js

```diff
@@ -0,0 +1,70 @@
+import { loadOneDriveAPI } from './onedrive-api-client';
+import { PickerCanceledError } from './google-picker-client';
+
+export class OneDrivePickerClient {
+  /**
+   * @param {object} options
+   * @param {string} options.clientId
+   * @param {string} options.redirectURI - the URL that is used to launch the picker
+   */
+  constructor({ clientId, redirectURI }) {
+    loadOneDriveAPI();
+
+    this._filePickerOptions = {
```

Comment:
This value is only used in one method and its structure is very specific to that use case. I would prefer this to be declared as a local there, with only the minimal amount of state (the `clientId` and `redirectURI` properties) kept as instance fields on the class.

Reply:
Implemented in the last commit.